db/migrations/25-03-07-compute-contribution-stats.sql

```diff
@@ -0,0 +1,133 @@
+begin;
+
+create table weekly_contribution_statistics (
+    id serial primary key,
+    week timestamp not null,
+    language_id uuid not null references language (id),
+    user_id uuid not null references users (id),
+    approved_count int not null,
+    revoked_count int not null,
+    edited_approved_count int not null,
+    edited_unapproved_count int not null,
+    unique (week, language_id, user_id)
+);
+
+create or replace function generate_weekly_contribution_stats(d timestamp)
+returns void
+language sql
+as $$
+    insert into weekly_contribution_statistics (
+        week,
+        language_id,
+        user_id,
+        approved_count,
+        revoked_count,
+        edited_approved_count,
+        edited_unapproved_count
+    )
+    select
+        date_bin('7 days', date_trunc('day', $1), timestamp '2024-12-15'),
+        changed_phrase.language_id,
+        changed_phrase.updated_by,
+        coalesce(count(*) filter (
+            where changed_phrase.state = 'APPROVED'
+                and coalesce(prev_phrase.state, 'UNAPPROVED') = 'UNAPPROVED'
+        ), 0) as words_approved,
+        coalesce(count(*) filter (
+            where changed_phrase.state = 'UNAPPROVED'
+                and coalesce(prev_phrase.state, 'UNAPPROVED') = 'APPROVED'
+        ), 0) as words_revoked,
+        coalesce(count(*) filter (
+            where changed_phrase.gloss <> coalesce(prev_phrase.gloss, '')
+                and changed_phrase.state = 'APPROVED'
+                and coalesce(prev_phrase.state, 'UNAPPROVED') = 'APPROVED'
+        ), 0) as words_approved_edited,
+        coalesce(count(*) filter (
+            where changed_phrase.gloss <> coalesce(prev_phrase.gloss, '')
+                and changed_phrase.state = 'UNAPPROVED'
+                and coalesce(prev_phrase.state, 'UNAPPROVED') = 'UNAPPROVED'
+        ), 0) as words_unapproved_edited
```

Comment:
Then we group by language and person, and count the number of glosses and gloss states that have changed from the previous week to now. This gets insert into a table as a row based on week, language, and translator